### PR TITLE
Issue #2827: JavadocMethod doesn't report anonymous class methods

### PR DESCRIPTION
**Pull request by AayushSaini101**
Related: https://github.com/checkstyle/checkstyle/issues/2827
Diff Regression config: https://gist.githubusercontent.com/AayushSaini101/218a9a57a04728fb154b406078059651/raw/9122a6c84272be78546541c4adf7ee3135d99b99/gistfile1.txt
Diff Regression Project:  https://gist.githubusercontent.com/AayushSaini101/0747edfd032791a6db97ab895e5c930c/raw/2b6a38b42b347c8827b7423cdd6086e40d95a236/gistfile1.txt

### CODE REVIEW

File: src/test/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/javadocmethod/InputJavadocMethodPublicOnly.java [hunk 1]
```diff
@@ -130,9 +130,6 @@ public void run() {
                 privateMethod(null, null);
             }
 
-            /**
-             * Javadoc
-             */
             private String privateMethod(String a, String b) {
```
Comment (romani) [inline src/test/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/javadocmethod/InputJavadocMethodPublicOnly.java:133]: Please do not update other Inputs with removal of code, they are weird by design.
You can put/remove violation comment only in them if required.

**Review by romani — CHANGES_REQUESTED** [review]
Please show regression diff report.
https://github.com/checkstyle/contribution/tree/master/checkstyle-tester#executing-generation
You can see bunch of examples in our merged PRs

Items:

File: src/test/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/javadocmethod/InputJavadocMethodPublicOnly.java (hunk 1 above)
Comment (AayushSaini101) [reply src/test/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/javadocmethod/InputJavadocMethodPublicOnly.java:133]: @romani I want to return three violations for the anonymous method. I am able to return one but unable to return 2. Can you help me 
   ```
public Thread anonymousClassInMethod() {
        return new Thread() {
            @Override
            public void run() {
                privateMethod(null,null);
            }

            /**
            * Javadoc
            */
            private String privateMethod(String a, String b) {   //  violation '@return tag should be present and have description.'
                return null;
            }
        };
    }
```

**Comment by AayushSaini101** [comment]
Github,  generate report

File: src/test/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/javadocmethod/InputJavadocMethodPublicOnly.java (hunk 1 above)
Comment (AayushSaini101) [reply src/test/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/javadocmethod/InputJavadocMethodPublicOnly.java:136]: > Please do not update other Inputs with removal of code, they are weird by design. You can put/remove violation comment only in them if required.

Done @romani

File: src/test/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/javadocmethod/InputJavadocMethodPublicOnly.java [hunk 2]
```diff
@@ -142,13 +145,16 @@ private String privateMethod(String a, String b) {
...
         }
 
         /**
          * Javadoc
          */
         public String publicMethod(String a, String b) {
+                 // violation above '@return tag should be present and have description.'
+                 // violation 2 lines above 'Expected @param tag for 'a'.'
+                // violation 3 lines above 'Expected @param tag for 'b'.'
             return null;
```
Comment (romani) [inline src/test/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/javadocmethod/InputJavadocMethodPublicOnly.java:158]: Vertically align comments with return.

File: src/test/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/javadocmethod/InputJavadocMethodPublicOnly.java [hunk 3]
```diff
@@ -142,13 +145,16 @@ private String privateMethod(String a, String b) {
     private final Thread anonymousClassInField = new Thread() {
         @Override
         public void run() {
-            publicMethod(null, null);
+            publicMethod(null,null);
```
Comment (romani) [inline src/test/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/javadocmethod/InputJavadocMethodPublicOnly.java:148]: Restore space.

**Review by romani — CHANGES_REQUESTED** [review]
You forgot to put link to list of properties to let action generate report 

Same for both input files

File: src/test/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/javadocmethod/InputJavadocMethodPublicOnly.java (hunk 3 above)
Comment (AayushSaini101) [reply src/test/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/javadocmethod/InputJavadocMethodPublicOnly.java]: Done @romani

File: src/test/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/javadocmethod/InputJavadocMethodPublicOnly.java (hunk 2 above)
Comment (AayushSaini101) [reply src/test/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/javadocmethod/InputJavadocMethodPublicOnly.java:158]: Done @romani

**Comment by AayushSaini101** [comment]
> You forgot to put link to list of properties to let action generate report
> 
> Same for both input files

Done @romani

**Comment by romani** [comment]
Provide `Diff Regression projects:`
See example https://github.com/checkstyle/checkstyle/pull/13953

And generate report again

File: src/test/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/javadocmethod/InputJavadocMethodPublicOnly1.java [hunk 4]
```diff
@@ -133,7 +133,7 @@ public void run() {
             /**
              * Javadoc
              */
-            private String privateMethod(String a, String b) {
+            private String privateMethod(String ax, String bx) {
```
Comment (romani) [inline src/test/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/javadocmethod/InputJavadocMethodPublicOnly1.java:136]: Please restore this code.
We do not change Inputs until absolutely required. All Input files are very weird and bad and this is exactly what we want

**Review by romani — APPROVED** [review]
Ok to merge

With minor comment to resolve before next review.

**Comment by romani** [comment]
@AayushSaini101 , please create PR by your  good account to show that all CIs are green

**Comment by AayushSaini101** [comment]
> @AayushSaini101 , please create PR by your good account to show that all CIs are green

Sure

File: src/test/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/javadocmethod/InputJavadocMethodPublicOnly1.java (hunk 4 above)
Comment (AayushSaini101) [reply src/test/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/javadocmethod/InputJavadocMethodPublicOnly1.java]: Done @romani

**Review by rnveach — CHANGES_REQUESTED** [review]
@romani This PR does not follow the issue description.
https://github.com/checkstyle/checkstyle/issues/2827#issuecomment-1507893087

> we need special property for this, not enabled by default. A lot of people will not like this.

**Comment by AayushSaini101** [comment]
> @romani This PR does not follow the issue description. [#2827 (comment)](https://github.com/checkstyle/checkstyle/issues/2827#issuecomment-1507893087)
> 
> > we need special property for this, not enabled by default. A lot of people will not like this.
Thanks @rnveach I will update it by using another property in the check

**Comment by AayushSaini101** [comment]
Closing this PR @romani @rnveach  I will use another account for all contribution

**Comment by romani** [comment]
>his PR does not follow the issue description.

Yes, you are right.
But it has very good step forward, as it demands javadoc comments to follow javadoc rules and it doesn't demand javadoc to be present if it absent.

So users who did not mean usage of javadoc but typed `/**` . Just need to change it to `/*`.

@AayushSaini101, please create new issue on this and use new issue in this commit.

**Comment by rnveach** [comment]
> as it demands javadoc comments to follow javadoc rules and it doesn't demand javadoc to be present if it absent

This does not follow the first post in the issue.

The expected output says that javadoc comments are wanted to be present, as they are currently not.

**Pull request closed without merging**